django_be/matchiq_api/api/data.py

```diff
@@ -0,0 +1,210 @@
+                    "type": "array",
+                    "description": "Other important information about the resume",
+                    "items": {"type": "string"},
+                },
+            },
+        }
+        gpt_res = self.gpt(system_prompt, user_prompt, schema)
+        res_json = json.loads(gpt_res)
+        return res_json
+
+
+    def get_score(self, resume_obj, job_obj):
+        system_prompt = "You are a job headhunter trying match people to specific jobs."
+        user_prompt = "Take in consideration the following resume and job posting. Keep in mind the experience, skills and education of the applicant. If they do not meet any qualifications, give them a 0 out of 100. Return a matching score 0 out of 100 for this candidate's resume compared to the following job posting: \n {}\n\n Job Posting:\n {}\n\n return the score and reasoning in a JSON format. Be brutal with the score. If they do not have any qualifications of the job, do not give them a good score.".format(
+            resume_obj, job_obj
+        )
+        schema = {
+            "type": "object",
+            "properties": {
+                "score": {
+                    "type": "string",
+                    "description": "A matching score out of 0 out of 100 that compares the applicant's qualifications to the job."
+
+                },
+                "reasoning": {
+                    "type": "string",
+                    "description": "Provide reasoning for the matching score out of 100.",
+                },
+            },
+        }
+        gpt_res = self.gpt(system_prompt, user_prompt, schema)
+        res_json = json.loads(gpt_res)
+        return res_json
+
+    # Params: job (obj)
+    # example_job = {
+    #     "title" : "Software Engineer",
+    #     "company" : "Netflix",
+    #     "description" : job_desc
+    # }
+    # resume_path = "path_to_resume_in_db.pdf"
+    def matching_score(self, job, resume_path):
+        resume_text = self.get_resume_text(resume_path)
+
+        # analyze both with gpt
+        # both are objects, so convert to string
+        print("Analyzing job...")
+        obj_job_desc = self.analyze_job_desc(job['description'])
+        obj_job_desc = json.dumps(obj_job_desc)
+
+        # sleep to avoid spamming openai
+        time.sleep(3)
+        print("Analyzing resume...")
+        obj_resume = self.analyze_resume(resume_text)
+        obj_resume = json.dumps(obj_resume)
+
+        time.sleep(3)
+        # get score with gpt
+        print("Getting matching score...")
+        matching_score = self.get_score(obj_resume, obj_job_desc)
+
+        return(matching_score)
+
+
+    # GPT Client, takes in system, user prompt and schema32                                                                     5t
+    def gpt(self, system_prompt, user_prompt, schema):
+        client = OpenAI()
+        completion = client.chat.completions.create(
+            model="gpt-3.5-turbo-0613",
```

Comment:
can we put the model as a const at the top? just so if we ever need to look up the model its right at the top